Implement unescape helper and style fixes.

mattcollier opened the pull request:
Although the test suite is passing, I don't know if this C++ code is doing the right thing, or if it's implemented properly, but I figured I'd get the ball rolling.

Just commented regex code so comparison would be easier, this stuff should be removed.

I found the benchmarks too variable to be useful, see: https://gist.github.com/mattcollier/33613cef6565c22cb78e84e20e34d260

Lifted the unescape code from here: https://stackoverflow.com/a/5612287

mattcollier committed:
Add .eslintrc.

mattcollier committed:
Fix style.

mattcollier committed:
Fix style.

mattcollier committed:
Fix style.

mattcollier committed:
Move .eslintrc.

mattcollier committed:
Implement unescape helper.

mattcollier committed:
Comment Urgna2012 benchmark.

mattcollier committed:
Add `test-suites` to .gitignore.

mattcollier committed:
Add windows newline.

mattcollier committed:
Add benchmark2.

mattcollier commented:
Added a benchmark that shows a 100% performance increase with the `unescape` implementation. 
 See results: https://gist.github.com/mattcollier/d01faf4e2f2be59f9d8796523e7f3a50

mattcollier committed:
Replace use of ostringstream.

mattcollier committed:
Add sync test.

dlongley commented:
@davidlehn, pulling this in would yield a very significant performance increase -- can you see if we have proper testing for it and add a test or two if not?

davidlehn commented on .eslintrc:
```diff
@@ -0,0 +1,10 @@
+  "env": {
+    "browser": true,
+    "mocha": true,
+    "phantomjs": true
```
This package isn't using phantomjs.
```suggestion
```

davidlehn commented on benchmark/benchmark.js:
```diff
@@ -37,16 +37,16 @@ if(!TEST_SUITE) {
+  //       format: 'application/nquads'
+  //     })
+  //   ]
+  // },
```
Is this supposed to be commented out?

davidlehn commented:
- I'm not sure what extra testing is needed.  I'm guessing it'd be some crazy escaping edge case. Would be in the main test suite if so.  Any thoughts on oddball patterns that could cause trouble?
- If we're not using regexes then can clean up that file: remove regex include, remove regexes, remove commented out code, move iostream header up, remove other unused commented old things.
- I think that regex call was the only reason we needed all that goofy travis compiler update code.  Would be nice to not need that!

mattcollier commented:
@davidlehn please make whatever changes are necessary here and wrap it up.

That commented about test suite stuff was not running properly, it doesn't seem to be a thing in the current test suites if you download a new copy of it.

davidlehn committed:
Remove phantomjs support.

davidlehn committed:
Use sibling jsonld.js.

Since benchmarking is a developer task, simplify installs, avoid some
potential dependency oddities, and just use a sibling directory for
jsonld.js.

davidlehn committed:
Canonicalization and optimization updates.

- Use simpler canonical N-Quads format.
  - Just escape 4 required chars.
- Remove regex and stream usage.
- Optimize native serialization by ~1-5x.
- Sync with rdf-canonize js code.

davidlehn committed:
Sync js with rdf-canonize.

davidlehn committed:
Simplify testing.

- Remove C++11 compiler upgrade.
  - Not required since regex_replace usage removed.
- Remove cruft.

dlongley commented on CHANGELOG.md:
```diff
@@ -1,5 +1,19 @@
+
+### Changed
+- N-Quad native serialization optimization.
+  - Varies based on input by rougly ~1-5x.
```
```suggestion
  - Varies based on input by roughly ~1-5x.
```

dlongley committed:
Fix typo.

Co-Authored-By: davidlehn <dil@lehn.org>

davidlehn committed:
Remove unused test dir option.

davidlehn committed:
Remove submodule.

davidlehn committed:
Update changelog.

davidlehn committed:
Add optimization TODO.
